code/Backend/flask-backend/app.py

```diff
@@ -380,71 +386,58 @@ def get_reservation():
     except Exception as e:
         return jsonify({"error": str(e)}), 500
 
-@app.route('/api/searchItems', methods=['GET'])
-def search_items():
-    try:
-        search_query = request.args.get('q')
-        print(f"Received search query: {search_query}")
 
+
+# Define the function to execute SQL queries and fetch data
+def execute_database_query(query, params=None):
+    try:
+        # Connect to the PostgreSQL database
         conn = psycopg2.connect(**db_connection_settings)
         cursor = conn.cursor()
-
-        query = "SELECT * FROM Equipment WHERE name ILIKE %s OR description ILIKE %s"
-        cursor.execute(query, (f"%{search_query}%", f"%{search_query}%"))
-
+        if params is not None:
+            cursor.execute(query, params)
+        else:
+            cursor.execute(query)
+        # Fetch all rows and store them in a list of dictionaries
         columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
+        data = [dict(zip(columns, row)) for row in cursor.fetchall()]
         cursor.close()
         conn.close()
-
-        if equipment_data:
-            return jsonify(equipment_data), 200
-        else:
-            return jsonify({"message": "No matching items found"}), 404
-
+        return data
     except Exception as e:
-        return jsonify({"error": str(e)}), 500
+        return None, str(e)
 
-@app.route('/api/availableItems', methods=['GET'])
-def get_available_items():
+@app.route('/api/searchItems', methods=['GET'])
+def search_items():
     try:
-        conn = psycopg2.connect(**db_connection_settings)
-        cursor = conn.cursor()
-
-        cursor.execute("SELECT * FROM Equipment WHERE available = true")
-        columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
-        cursor.close()
-        conn.close()
 
+        search_query = request.args.get('q')
+        print(f"Received search query: {search_query}")
+        query = "SELECT * FROM Equipment WHERE name ILIKE %s OR description ILIKE %s"
```

Comment:
Not sure why you kept using literal string for the query here when you have created query constant on line 34?

Reply:
I forgot to update my backend code for that, I had been juggling trying to get Selenium to work, and finish up researching how to pass the item to the Reservation Page I Missed to make the change